### PR TITLE
Fix issue #944

### PR DESCRIPTION
**Pull request by Dissfall**
Fixed a bug where it was impossible to detect devices controlled by NetworkManager on some Russian-language systems

### CODE REVIEW

**Comment by sophron** [comment]
Hi @Dissfall and thank you for contributing.

We just recently started supporting `nmcli` tool and we expect to have introduced some bugs, like the one that you discovered. While your proposed solution will fix the issue for Russian locales, it won't work for other languages. At the same time, having to support all languages (e.g. by having the "unamanaged" string in each language) will be an operational overhead that we want to avoid.

What I recommend is to check the locale like you do and if it is not English, then skip the check `if state_unmanaged not in l` at all. This has the obvious risk that the we may fail to check if the interface was successfully set to unmanaged in systems with a language set other than English but we can potentially catch this exception in a later phase.

**Comment by blackHatMonkey** [comment]
@sophron I agree that this is not a proper fix for the problem. Perhaps in the long run we can internationalize and add more locales with `gettext` module. This will allow for crowd sourcing translation and increase the popularity of the tool.

**Comment by Dissfall** [comment]
@sophron @blackHatMonkey I completely agree with you. My solution is not the most effective and does not work with other languages. I like @blackHatMonkey proposal for using the `gettext` module. I'll try to make a more elegant decision ...

**Comment by sophron** [comment]
@blackHatMonkey 

`gettext` is an internationalization tool that we may use in order to translate the entire application globally. By translating the entire application, we basically mean to translate the *output* of the software, or in other words, what the users see. I'm fine with introducing `gettext` at some point.

However, we should not use `gettext` for translating any input of the application though. We should not depend on a "translation" for defining the program's behavior as in this case.

File: wifiphisher/common/interfaces.py [hunk 1]
```diff
@@ -834,13 +838,26 @@ def is_managed_by_network_manager(interface_name):
...
     try:
-        out = check_output(["nmcli", "dev"])
-        for l in out.splitlines():
-            #TODO: If the device is managed and user has nmcli installed, 
-            # we can probably do a "nmcli dev set wlan0 managed no"
-            if interface_name in l and "unmanaged" not in l:
-                is_managed = True
-    # NetworkManager service is not running so the devices must be unmanaged 
+        nmcli_process = Popen(['/bin/sh', '-c', 'export LC_ALL=C; nmcli dev; unset LC_ALL'], stdout=PIPE);
+        out, err = nmcli_process.communicate();
```
[the remark this answers is not in the document]
Comment (Dissfall) [reply wifiphisher/common/interfaces.py]: I tried to use that function. `check_output()` not support native BASH methods like `export`

File: wifiphisher/common/interfaces.py [hunk 2]
```diff
@@ -834,13 +838,26 @@ def is_managed_by_network_manager(interface_name):
...
+
+        if err == None:
+            for l in out.splitlines():
+                #TODO: If the device is managed and user has nmcli installed,
+                # we can probably do a "nmcli dev set wlan0 managed no"
+                if interface_name in l:
+                    if interface_name in l and "unmanaged" not in l:
```
[the remark this answers is not in the document]
Comment (Dissfall) [reply wifiphisher/common/interfaces.py]: Yes, I encountered an error when the command returned an empty string. A more correct and risk-free version would look something like this:
```
...
if err == None and out != "":
            for l in out.splitlines():
                #TODO: If the device is managed and user has nmcli installed,
                # we can probably do a "nmcli dev set wlan0 managed no"
                if interface_name in l:
                    if "unmanaged" not in l:
                        is_managed = True
                else:
                    logger.error('Error obtaining base process output')
                    logger.error('Ignored checking for managed devices by NM')
        else:
            logger.error('Error opening the base process')
            logger.error('Ignored checking for managed devices by NM')
...
```

**Comment by blackHatMonkey** [comment]
@sophron Instead of checking the output to see if an interface is managed or not maybe we can just try to unmange it instead so it would be something like this:
```python
subprocess.check_call(
            ["nmcli", "dev", "set", interface_name, "managed", "no"],
            stdout=constants.DN,
            stderr=constants.DN)
```

**Comment by Dissfall** [comment]
Something strange happened in my copy of the repository and I can not fix it(

**Comment by Dissfall** [comment]
Ok. I create new RP and make pull request https://github.com/wifiphisher/wifiphisher/pull/945

**Pull request closed without merging**